### PR TITLE
feat: Add shorts embedding.

### PR DESCRIPTION
**Pull request by dohooo**
### WHAT

copilot:summary

copilot:poem

### WHY

https://github.com/Crossbell-Box/xLog/assets/32405058/1cc3c17f-cfda-44af-97b7-f38a889c849e

https://github.com/Crossbell-Box/xLog/assets/32405058/6a061680-0c5a-4626-972f-360004ab071c

https://github.com/Crossbell-Box/xLog/assets/32405058/66cb496c-31dc-4234-9ad9-fa4fb48ad06c

Screenshot 2023-12-20 at 13 12 12

### HOW

copilot:walkthrough

### CLAIM REWARDS

For first-time contributors, please leave your xLog address and Discord ID below to claim your rewards.

### CODE REVIEW

**Review by DIYgod — CHANGES_REQUESTED** [review]

File: src/markdown/embed-transformers/XLogShorts.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import type { Transformer } from "../rehype-embed"
+import { isHostIncludes } from "./utils"
+
+export const XLogShortsTransformer: Transformer = {
+  name: "XLogShorts",
+  shouldTransform(url) {
+    // There are two patterns for xlog shorts url
+
+    const { host, pathname, searchParams } = url
+    const hasShortsParam = searchParams.get("ct") === "shorts"
```
[the remark this answers is not in the document]
Comment (dohooo) [reply src/markdown/embed-transformers/XLogShorts.ts:12]: I want to make sure the old articles are not affected.

File: src/components/ui/XLogShorts.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  const toggleExpand = useCallback(() => {
+    setIsExpanded((prev) => !prev)
+  }, [])
+
+  if (page.isLoading) return <div>Loading...</div>
+
+  return (
+    <div
+      onClick={toggleExpand}
```
[the remark this answers is not in the document]
Comment (dohooo) [reply src/components/ui/XLogShorts.tsx:43]: OK

File: src/markdown/embed-transformers/XLogShorts.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (DIYgod) [reply src/markdown/embed-transformers/XLogShorts.ts:12]: It's okay, I haven't seen anyone put shorts links in an article yet, and it needs to be put as a separate paragraph.

[the remark this answers is not in the document]
Comment (dohooo) [reply src/markdown/embed-transformers/XLogShorts.ts:12]: Okay

[the remark this answers is not in the document]
Comment (dohooo) [reply src/markdown/embed-transformers/XLogShorts.ts:12]: ### Not shorts content.
image

### Shorts content
image

**Pull request merged**